app/controllers/find/application_controller.rb

```diff
@@ -16,11 +16,23 @@ def render_feedback_component
       @render_feedback_component = true
     end
 
+    def backlink_query_parameters
+      ResultsView.new(
+        query_parameters: request.query_parameters[form_name].presence || request.query_parameters
+      )
+                 .query_parameters_with_defaults
+                 .except(form_name)
+    end
+
     # DFE Analytics namespace
     def current_namespace
       'find'
     end
 
+    def undergraduate_feature_enabled?
+      Settings.current_recruitment_cycle_year.to_i >= 2025 && FeatureService.enabled?(:teacher_degree_apprenticeship)
```

Comment:
Actually they are slightly different. For Find we want to show only for the next cycle while the method above will look for the `course/provider.recruitment_cycle. recruitment_cycle_year` to check (then we show before the cycle starts).

Also on Publish we should use from `Course`. On Find I don't have the scope of the course since we are searching many courses.

Reply:
I agree with naming, I changed to `teacher_degree_apprenticeship_active?`